Sources/Log.cpp

```diff
@@ -0,0 +1,38 @@
+#include "Log.h"
+
+QFile logs("log-"+ QDateTime::currentDateTime().toString() + "-" + QUuid::createUuid().toString() + ".txt");
+QTextStream logstream(&logs);
+
+void InformationToStream(const QString type, const QMessageLogContext &context, const QString &message, const QString &datetime)
+{
+    logstream << type << datetime << ": " << message << " " << context.file << " " << context.line << " " << context.function << endl;
```

Comment:
File, line and function should be before message, because otherwise it will be uncomfortable to read logs.